.github/workflows/ci.yaml

```
@@ -28,11 +28,9 @@ jobs:
     - name: Update Cabal's database
       run: nix-shell --pure --run "cabal update"
     - name: Build Cabal's dependencies
-      run: nix-shell --pure --run "cabal build --enable-tests --enable-benchmarks --dependencies-only"
+      run: nix-shell --pure --run "cabal build --disable-tests --disable-benchmarks --dependencies-only"
     - name: Build
-      run: nix-shell --pure --run "cabal build"
-    - name: Test
-      run: nix-shell --pure --run "cabal test"
+      run: nix-shell --pure --run "cabal build --disable-tests --disable-benchmarks"
```

You don't have to disable tests/benchmarks again (in fact, I believe it's the default anyway, that they are disabled).
```suggestion
      run: nix-shell --pure --run "cabal build"
```

---

Well, I think that they are enabled by default: see this CI job that failed on friday when I didn't explicitely disabled them: https://github.com/tweag/linear-base/actions/runs/1969067243 and https://github.com/tweag/linear-base/commit/e64d220f764f7074e9e0f20cbafdeb8ff0d1a0a8

---

That's because they've been enabled above. the `--enable-tests` mutates a state. It doesn't tell `cabal` to run tests. Try with the suggestions applied, and we will have a definite answer to that question though.

---

Cabal CLI is stateful? :scream: